packages/@webex/internal-plugin-scheduler/src/scheduler/scheduler.js

```diff
@@ -0,0 +1,156 @@
+import {WebexPlugin} from '@webex/webex-core';
+
+import CONSTANTS from './scheduler.constants';
+
+/**
+ * Scheduler WebexPlugin class.
+ */
+class Scheduler extends WebexPlugin {
+  /**
+   * Namespace, or key, to register a `Scheduler` class object to within the
+   * `webex.internal` object. Note that only one instance of this class can be
+   * used against a single `webex` class instance.
+   */
+  namespace = CONSTANTS.NAMESPACE;
+
+  /**
+   * @param {Object} args Arguments for constructing a new Scheduler instance.
+   */
+  constructor(...args) {
+    super(...args); // Required to properly mount the singleton class instance.
+
+    /**
+     * The `this.request` method should now be available for usage.
+     * See `http-core` for this tooling.
+     */
+
+    /**
+     * The mercury connection is available for initialization
+     */
+
+    /**
+     * The `this.logger` object should now be available for usage.
+     * `this.logger.log()`, `this.logger.error()`, `this.logger.warn()`, etc.
+     */
+    this.logger.log('plugin example constructed'); // example, remove this.
+  }
+
+  /**
+   * WebexPlugin initialize method. This triggers once Webex has completed its
+   * initialization workflow.
+   *
+   * If the plugin is meant to perform startup actions, place them in this
+   * `initialize()` method instead of the `constructor()` method.
+   * @returns {void}
+   */
+  initialize() {
+    // Used to perform actions based on the provided configuration object once
+    // the configuration object is ready.
+    this.listenToOnce(this.webex, 'change:config', () => {
+      /* ...perform actions once the configuration object is mounted... */
+    });
+
+    // Used to perform actions after webex is fully qualified and ready for
+    // operation.
+    this.listenToOnce(this.webex, 'ready', () => {
+      /* ...perform actions once the webex object is fully qualified... */
+    });
+  }
+
+  /**
+   * Example request usage. `this.request` returns a promise using XHR. and the
+   * `request` npm package.
+   * @returns {void}
+   */
+  exampleRequestUsage() {
```

Comment:
can you move this stuff into some sub folders as examples as well? it doesn't all need to be done in the scheduler.js file

Reply:
I considered this, and spent some time writing some templates like this. But, decided ultimately against doing this, as the architect/director of this plugin should have the freedom to adjust folder structure, relations between models, and event pipelines as needed.

I will stand by for QA for best practices and general inquiries from the team owning this plugin.